Refactor DDP checkpoint tests

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68827
* __->__ #68792

Refactor tests to be more clear what features are supported and
unsupported under certain DDP configs.

Differential Revision: [D32609498](https://our.internmc.facebook.com/intern/diff/D32609498/)

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

rohan-varma committed:
Refactor DDP checkpoint tests

Refactor tests to be more clear what features are supported and
unsupported under certain DDP configs.

Differential Revision: [D32609498](https://our.internmc.facebook.com/intern/diff/D32609498/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Refactor DDP checkpoint tests"


Refactor tests to be more clear what features are supported and
unsupported under certain DDP configs.

Differential Revision: [D32609498](https://our.internmc.facebook.com/intern/diff/D32609498/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

pritamdamania87 commented on test/distributed/test_c10d_nccl.py:
```diff
@@ -2136,16 +2170,9 @@ def test_ddp_checkpointing_unused_params(self):
                     find_unused_parameters=True,
                     static_graph=False,
```
nit: we can omit `static_graph=False`?

pritamdamania87 commented on test/distributed/test_c10d_nccl.py:
```diff
@@ -2162,14 +2189,29 @@ def test_ddp_checkpointing_twice(self):
+                    self.CheckpointTwiceModule(),
+                    process_group=process_group,
+                    use_bucket_view=use_bucket_view,
+                    static_graph=False,
```
How does this work with static_graph=False?

rohan-varma replied:
It doesn't, if you expand the code above this is in a `assertRaisesRegex` block. This particular test is verifying with static_graph=False, and for both unused param settings, the checkpointing twice fails because we expect to mark a variable ready only once.

rohan-varma committed:
Update on "Refactor DDP checkpoint tests"


Refactor tests to be more clear what features are supported and
unsupported under certain DDP configs.

Differential Revision: [D32609498](https://our.internmc.facebook.com/intern/diff/D32609498/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

pritamdamania87 commented on test/distributed/test_c10d_nccl.py:
```diff
@@ -2160,16 +2186,29 @@ def test_ddp_checkpointing_twice(self):
                     use_bucket_view=use_bucket_view,
-                    static_graph=False,
                 )
+                model = self._test_ddp_checkpointing(
```
Shouldn't this be under its own `assertRaisesRegex` block? I don't think this line would get executed since the previous line itself would throw an exception?

rohan-varma committed:
Update on "Refactor DDP checkpoint tests"


Refactor tests to be more clear what features are supported and
unsupported under certain DDP configs.

Differential Revision: [D32609498](https://our.internmc.facebook.com/intern/diff/D32609498/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

rohan-varma committed:
Update on "Refactor DDP checkpoint tests"


Refactor tests to be more clear what features are supported and
unsupported under certain DDP configs.

Differential Revision: [D32609498](https://our.internmc.facebook.com/intern/diff/D32609498/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

zhaojuanmao reviewed (approved):
nice!

rohan-varma committed:
Update on "Refactor DDP checkpoint tests"


Refactor tests to be more clear what features are supported and
unsupported under certain DDP configs.

Differential Revision: [D32609498](https://our.internmc.facebook.com/intern/diff/D32609498/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]
